### PR TITLE
tests(e2e): add cert-manager test

### LINKED ISSUE #1734: add e2e test for cert-manager
This issue is to add e2e tests around using cert-manager in osm.

### PR DESCRIPTION
**Pull request by nojnhuh**
**Description**:

This change adds a simple e2e test with one pod trying to reach one
server using cert-manager to issue certificates.

Fixes #1734

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [X]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: tests/e2e/e2e_certmanager_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+var _ = Describe("1 Client pod -> 1 Server pod test using cert-manager", func() {
+	Context("CertManagerSimpleClientServer", func() {
+		const sourceNs = "client"
+		const destNs = "server"
+		var ns []string = []string{sourceNs, destNs}
+
+		It("Tests HTTP traffic for client pod -> server pod", func() {
+			// Install OSM
+			installOpts := td.GetOSMInstallOpts()
+			installOpts.certManager = "cert-manager"
```
Comment (eduser25) [inline tests/e2e/e2e_certmanager_test.go:20]: I'd suggest to make these an enum at common.go at this point

**Review by eduser25 — APPROVED** [review]

File: tests/e2e/common.go [hunk 2]
```diff
@@ -437,6 +460,103 @@ tail /dev/random;
 	return nil
 }
 
+func (td *OsmTestData) installCertManager(instOpts InstallOSMOpts) error {
+	By("Installing cert-manager")
```
Comment (eduser25) [inline tests/e2e/common.go:464]: somehow related, I'm scared of `common.go` growing with no boundaries. problem of always using _helpers.go_ or _common.go_ or _utils.go_...

File: tests/e2e/e2e_certmanager_test.go (hunk 1 above)
Comment (nojnhuh) [reply tests/e2e/e2e_certmanager_test.go:20]: I'd vote to skip that step and make it an enum somewhere in `pkg/` that we can just import here.

File: tests/e2e/common.go (hunk 2 above)
Comment (nojnhuh) [reply tests/e2e/common.go:464]: The file becoming too big is easy to fix, but the harder problem would be controlling the growth of the number of methods on `OsmTestData`.

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**